packages/fuels-abigen-macro/tests/test_projects/enums_with_unit_types/src/main.sw

```diff
@@ -0,0 +1,24 @@
+contract;
+
+enum OnlyUnitsInEnum {
+    a: (),
+    b: (),
+}
+
```

Comment:
This test project should be merged with the `use_enum_input` test (there might be some necessary refactoring)

Reply:
I've joined this one and the 'use_enum_input' into a new one that has more robust checks in place.

I've found that if you just return what you received as an arg you can have improperly encoded enums just returned to you and then you wouldn't test anything since you'd be in an echo chamber. This approach fixes that.